llvm/lib/DebugInfo/Symbolize/Symbolize.cpp

```diff
@@ -662,50 +662,64 @@ LLVMSymbolizer::getOrCreateModuleInfo(StringRef ModuleName) {
   ObjectPair Objects = ObjectsOrErr.get();
 
   std::unique_ptr<DIContext> Context;
-  // If this is a COFF object containing PDB info and not containing DWARF
-  // section, use a PDBContext to symbolize. Otherwise, use DWARF.
-  // Create a DIContext to symbolize as follows:
-  // - If there is a GSYM file, create a GsymContext.
-  // - Otherwise, if this is a COFF object containing PDB info, create a
-  // PDBContext.
-  // - Otherwise, create a DWARFContext.
-  const auto GsymFile = lookUpGsymFile(BinaryName.str());
-  if (!GsymFile.empty()) {
-    auto ReaderOrErr = gsym::GsymReader::openFile(GsymFile);
-
-    if (ReaderOrErr) {
-      std::unique_ptr<gsym::GsymReader> Reader =
-          std::make_unique<gsym::GsymReader>(std::move(*ReaderOrErr));
-
-      Context = std::make_unique<gsym::GsymContext>(std::move(Reader));
+  pdb::PDB_ReaderType ReaderType =
+      Opts.UseDIA ? pdb::PDB_ReaderType::DIA : pdb::PDB_ReaderType::Native;
+  const auto *CoffObject = dyn_cast<COFFObjectFile>(Objects.first);
+
+  // First, if the user specified a pdb file on the command line, use that.
+  if (CoffObject && !Opts.PDBName.empty()) {
+    using namespace pdb;
+    std::unique_ptr<IPDBSession> Session;
+    if (auto Err = loadDataForPDB(ReaderType, Opts.PDBName, Session)) {
+      Modules.emplace(ModuleName, std::unique_ptr<SymbolizableModule>());
+      return createFileError(Opts.PDBName, std::move(Err));
     }
+    Context.reset(new PDBContext(*CoffObject, std::move(Session)));
   }
```

Comment:
This is a duplicate of the code creating the PDB context below. Could these be merged?